cephfs/userperm_test.go

```diff
@@ -0,0 +1,43 @@
+package cephfs
+
+import (
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUserPerm(t *testing.T) {
+	t.Run("newAndDestroy", func(t *testing.T) {
+		uperm := NewUserPerm(0, 0, nil)
+		assert.NotNil(t, uperm)
+		assert.Equal(t, 0, len(uperm.gidList))
+		assert.True(t, uperm.managed)
+
+		// Destroy should be idempotent in our go library
+		uperm.Destroy()
```

Comment:
there is a potential race, not sure it is worth adding a mutex/lock to the UserPerm type though.

Reply:
I think that should be up to the caller, and in most cases I don't think it would be needed.